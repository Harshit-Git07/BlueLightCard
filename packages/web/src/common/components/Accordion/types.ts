export interface AccordionProps {
  title: string;
  children: React.ReactNode;
  onClickOpen?: () => void;
}
