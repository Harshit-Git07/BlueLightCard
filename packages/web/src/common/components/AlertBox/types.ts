export interface AlertBoxProps {
  alertType: string;
  title: string;
  description: any;
}

export interface style {
  textColor: string;
  backgroundColor: string;
}

export interface config {
  [key: string]: style;
}
