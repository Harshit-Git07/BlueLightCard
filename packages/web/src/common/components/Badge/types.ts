export type BgColorString = `bg-${string}`;

export type BadgeProps = {
  label: string;
  color: BgColorString;
  size: 'small' | 'large';
};
