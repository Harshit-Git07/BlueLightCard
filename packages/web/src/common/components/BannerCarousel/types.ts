import { CampaignCardProps } from '../CampaignCard/types';

export type BannerCarouselPropers = {
  banners: CampaignCardProps[];
};
