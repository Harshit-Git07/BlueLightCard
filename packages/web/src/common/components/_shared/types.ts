import { IconDefinition } from '@fortawesome/fontawesome-svg-core';
import { ChangeEventHandler, ForwardedRef, MouseEventHandler, PropsWithChildren } from 'react';

export type InputFieldWrapperProps = PropsWithChildren & {
  icon?: IconDefinition;
  iconRight?: IconDefinition;
  showRightIcon?: boolean;
  showErrorState?: boolean;
  select?: boolean;
  showSuccessState?: boolean;
  onRightIconClick?: MouseEventHandler;
};

export interface InputFieldSharedProps<E, V = string> {
  name?: string;
  error?: boolean;
  success?: boolean;
  value?: V;
  required?: boolean;
  _ref?: ForwardedRef<unknown>;
  onChange?: ChangeEventHandler<E>;
}
