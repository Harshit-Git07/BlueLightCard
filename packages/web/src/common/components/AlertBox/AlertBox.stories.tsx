import React from 'react';
import { Meta, StoryFn } from '@storybook/react';
import AlertBox from './AlertBox';
import { AlertBoxProps } from './types';

export default {
  title: 'Component System/AlertBox',
  component: AlertBox,
} as Meta;

const Template: StoryFn<AlertBoxProps> = (args) => <AlertBox {...args} />;

export const SuccessAlert = Template.bind({});
SuccessAlert.args = {
  alertType: 'success',
  title: 'Success Alert',
  description: 'This is a success message.',
};

export const WarningAlert = Template.bind({});
WarningAlert.args = {
  alertType: 'warning',
  title: 'Warning Alert',
  description: 'This is a warning message.',
};

export const ErrorAlert = Template.bind({});
ErrorAlert.args = {
  alertType: 'danger',
  title: 'Error Alert',
  description: 'This is an error message.',
};

export const InfoAlert = Template.bind({});
InfoAlert.args = {
  alertType: 'info',
  title: 'Info Alert',
  description: 'This is an info message.',
};
