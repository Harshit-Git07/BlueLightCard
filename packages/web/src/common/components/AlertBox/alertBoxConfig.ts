import { config } from './types';

const alertBoxConfig: config = {
  success: {
    textColor: 'text-palette-success',
    backgroundColor: 'bg-[#d4edda]',
  },
  danger: {
    textColor: 'text-palette-danger',
    backgroundColor: 'bg-[#f1b0b7]',
  },
  warning: {
    textColor: 'text-palette-warning',
    backgroundColor: 'bg-[#eb9316]',
  },
  info: {
    textColor: 'text-palette-info',
    backgroundColor: 'bg-[#3cf]',
  },
};

export default alertBoxConfig;
