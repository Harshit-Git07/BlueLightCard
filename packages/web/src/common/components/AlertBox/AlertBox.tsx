import { FC, useState } from 'react';
import alertBoxConfig from './alertBoxConfig';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faXmark } from '@fortawesome/pro-solid-svg-icons';
import { AlertBoxProps } from './types';

const AlertBox: FC<AlertBoxProps> = ({ alertType, title, description }) => {
  const [show, setShow] = useState(true);

  const { textColor, backgroundColor } = alertBoxConfig[alertType];

  return (
    <div
      className={`${backgroundColor} ${textColor} w-full py-[0.75rem] px-[2rem] flex justify-between m-2 rounded-4 ${
        !show && 'hidden'
      }`}
    >
      <div className="flex">
        <p>
          <strong>{title}</strong>
        </p>
        <p className="pl-1">{description}</p>
      </div>

      <button onClick={() => setShow(false)}>
        <FontAwesomeIcon icon={faXmark} />
      </button>
    </div>
  );
};

export default AlertBox;
