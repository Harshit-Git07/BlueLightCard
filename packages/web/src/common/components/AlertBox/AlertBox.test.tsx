import AlertBox from '@/components/AlertBox/AlertBox';
import { AlertBoxProps } from '@/components/AlertBox/types';
import renderer from 'react-test-renderer';

describe('AlertBox component', () => {
  it('should render success alert', () => {
    const props: AlertBoxProps = {
      alertType: 'success',
      title: 'Success Alert',
      description: 'This is a success message.',
    };

    const component = renderer.create(<AlertBox {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render info alert', () => {
    const props: AlertBoxProps = {
      alertType: 'info',
      title: 'Info Alert',
      description: 'This is an info message.',
    };

    const component = renderer.create(<AlertBox {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render warning alert', () => {
    const props: AlertBoxProps = {
      alertType: 'warning',
      title: 'Warning Alert',
      description: 'This is a warning message.',
    };

    const component = renderer.create(<AlertBox {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render danger alert', () => {
    const props: AlertBoxProps = {
      alertType: 'danger',
      title: 'Error Alert',
      description: 'This is an error message.',
    };

    const component = renderer.create(<AlertBox {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
