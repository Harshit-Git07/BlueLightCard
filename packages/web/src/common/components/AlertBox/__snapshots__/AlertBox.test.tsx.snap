// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertBox component should render danger alert 1`] = `
<div
  className="bg-[#f1b0b7] text-palette-danger w-full py-[0.75rem] px-[1.5rem] flex justify-between rounded-4 false"
>
  <div
    className="flex"
  >
    <p>
      <strong>
        Error Alert
      </strong>
    </p>
    <p
      className="pl-1"
    >
      This is an error message.
    </p>
  </div>
  <button
    onClick={[Function]}
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-xmark "
      data-icon="xmark"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={{}}
      viewBox="0 0 384 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M342.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 210.7 86.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L146.7 256 41.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192 301.3 297.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.3 256 342.6 150.6z"
        fill="currentColor"
        style={{}}
      />
    </svg>
  </button>
</div>
`;

exports[`AlertBox component should render info alert 1`] = `
<div
  className="bg-[#3cf] text-palette-info w-full py-[0.75rem] px-[1.5rem] flex justify-between rounded-4 false"
>
  <div
    className="flex"
  >
    <p>
      <strong>
        Info Alert
      </strong>
    </p>
    <p
      className="pl-1"
    >
      This is an info message.
    </p>
  </div>
  <button
    onClick={[Function]}
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-xmark "
      data-icon="xmark"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={{}}
      viewBox="0 0 384 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M342.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 210.7 86.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L146.7 256 41.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192 301.3 297.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.3 256 342.6 150.6z"
        fill="currentColor"
        style={{}}
      />
    </svg>
  </button>
</div>
`;

exports[`AlertBox component should render success alert 1`] = `
<div
  className="bg-[#d4edda] text-palette-success w-full py-[0.75rem] px-[1.5rem] flex justify-between rounded-4 false"
>
  <div
    className="flex"
  >
    <p>
      <strong>
        Success Alert
      </strong>
    </p>
    <p
      className="pl-1"
    >
      This is a success message.
    </p>
  </div>
  <button
    onClick={[Function]}
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-xmark "
      data-icon="xmark"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={{}}
      viewBox="0 0 384 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M342.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 210.7 86.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L146.7 256 41.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192 301.3 297.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.3 256 342.6 150.6z"
        fill="currentColor"
        style={{}}
      />
    </svg>
  </button>
</div>
`;

exports[`AlertBox component should render warning alert 1`] = `
<div
  className="bg-[#eb9316] text-palette-warning w-full py-[0.75rem] px-[1.5rem] flex justify-between rounded-4 false"
>
  <div
    className="flex"
  >
    <p>
      <strong>
        Warning Alert
      </strong>
    </p>
    <p
      className="pl-1"
    >
      This is a warning message.
    </p>
  </div>
  <button
    onClick={[Function]}
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-xmark "
      data-icon="xmark"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={{}}
      viewBox="0 0 384 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M342.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L192 210.7 86.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L146.7 256 41.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L192 301.3 297.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L237.3 256 342.6 150.6z"
        fill="currentColor"
        style={{}}
      />
    </svg>
  </button>
</div>
`;
